### PR TITLE
system-variables: Add charset documentation

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Add documentation for these:
<pre><span style="background-color:#E4E4E4"><font color="#005F5F">5.7.25-TiDB-v5.2.0-alpha-164-g8434069c5</font></span><span style="background-color:#3A3A3A"><font color="#E4E4E4"></font></span><span style="background-color:#3A3A3A"><font color="#FDF6E3"> 127.0.0.1:4000 </font></span><span style="background-color:#6C6C6C"><font color="#3A3A3A"></font></span><span style="background-color:#6C6C6C"><font color="#FDF6E3"> test </font></span><span style="background-color:#D75F00"><font color="#6C6C6C"></font></span><span style="background-color:#D75F00"><font color="#FDF6E3"> SQL </font></span><span style="background-color:#073642"><font color="#D75F00"> </font></span>show global variables like &apos;char%&apos;;
+--------------------------+--------------------------------------------------------+
| Variable_name            | Value                                                  |
+--------------------------+--------------------------------------------------------+
| character_set_client     | utf8mb4                                                |
| character_set_connection | utf8mb4                                                |
| character_set_database   | utf8mb4                                                |
| character_set_filesystem | binary                                                 |
| character_set_results    | utf8mb4                                                |
| character_set_server     | utf8mb4                                                |
| character_set_system     | utf8                                                   |
| character_sets_dir       | /usr/local/mysql-5.6.25-osx10.8-x86_64/share/charsets/ |
+--------------------------+--------------------------------------------------------+
8 rows in set (0.0042 sec)
</pre>

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

Related to  #3155

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -84,6 +84,54 @@ mysql> SELECT * FROM t1;
...
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- This indicates the character set of the default database that is in use. Setting this variable is not recommended. The server changes this varible when a new default database is selected.
+
+### character_set_filesystem
+
+- Scope: SESSION | GLOBAL
+- Default value: `binary`
+- This variable is not used and exists for compatibility with MySQL.
```
Comment (morgo) [inline system-variables.md:109]: We don't currently document any of the noop sysvars. I suggest we omit these, as I hope that in the next sprint we will also offer a feature to disable them (if it doesn't go ahead, we can maybe revisit them after all the actual sysvars are documented.)

Comment (dveeden) [reply system-variables.md:109]: Does `CharacterSetFilesystem` fall into that category?

It is defined in `sysvar.go` and not in `noop-.go` like `CharacterSetSystem`.

https://github.com/pingcap/tidb/blob/ac5cf972159822928b3e9ee1b848e7f4fe0a8d0d/sessionctx/variable/sysvar.go#L682-L683

https://github.com/pingcap/tidb/blob/ac5cf972159822928b3e9ee1b848e7f4fe0a8d0d/sessionctx/variable/noop.go#L157

Comment (morgo) [reply system-variables.md:109]: Yes. It looks like it should actually be in `noop.go`.

Comment (dveeden) [reply system-variables.md:109]: Ok. I'll remove variables defined in `noop.go` from this.
I'll also remove `character_set_filesystem`.

File: system-variables.md [hunk 2]
```diff
@@ -84,6 +84,36 @@ mysql> SELECT * FROM t1;
...
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- The character set that is used when data is sent to the client.
+
+### character_set_server
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- The collation used for new schemas in case none is provided by the `CREATE SCHEMA` statement.
```
Comment (bb7133) [inline system-variables.md:115]:
```suggestion
- The character set used for new schemas in case none is provided by the `CREATE SCHEMA` statement.
```

**Review by morgo — APPROVED** [review]

**Review by bb7133 — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -84,6 +84,36 @@ mysql> SELECT * FROM t1;
 - Default value: `ON`
 - Controls whether statements should automatically commit when not in an explicit transaction. See [Transaction Overview](/transaction-overview.md#autocommit) for more information.
 
+### character_set_client
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- Character set for data send by the client. See [Character Set and Collation](/character-set-and-collation.md) for details on the use of character sets and collations in TiDB. It is recommended to use [`SET NAMES`](/sql-statements/sql-statement-set-names.md) to change the character set when needed.
```
Comment (TomShawn) [inline system-variables.md:91]:
```suggestion
- The character set for data sent from the client. See [Character Set and Collation](/character-set-and-collation.md) for details on the use of character sets and collations in TiDB. It is recommended to use [`SET NAMES`](/sql-statements/sql-statement-set-names.md) to change the character set when needed.
```

File: system-variables.md [hunk 4]
```diff
@@ -84,6 +84,36 @@ mysql> SELECT * FROM t1;
...
+
+### character_set_connection
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- Character set for string literals that don't have a character set specified.
```
Comment (TomShawn) [inline system-variables.md:97]:
```suggestion
- The character set for string literals that do not have a specified character set.
```

File: system-variables.md [hunk 5]
```diff
@@ -84,6 +84,36 @@ mysql> SELECT * FROM t1;
...
+
+### character_set_database
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- This indicates the character set of the default database that is in use. Setting this variable is not recommended. The server changes this varible when a new default database is selected.
```
Comment (TomShawn) [inline system-variables.md:103]:
```suggestion
- This variable indicates the character set of the default database in use. **It is NOT recommended to set this variable**. When a new default database is selected, the server changes the variable value.
```

File: system-variables.md [hunk 6]
```diff
@@ -84,6 +84,36 @@ mysql> SELECT * FROM t1;
...
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- The character set that is used when data is sent to the client.
+
+### character_set_server
+
+- Scope: SESSION | GLOBAL
+- Default value: `utf8mb4`
+- The character set used for new schemas in case none is provided by the `CREATE SCHEMA` statement.
```
Comment (TomShawn) [inline system-variables.md:115]:
```suggestion
- The character set used for new schemas when no character set is specified in the `CREATE SCHEMA` statement.
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**